AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Actions/SignOut/ShowHostedUISignOut.swift

```diff
@@ -0,0 +1,128 @@
+            let event = SignOutEvent(eventType: .signedOutFailure(error))
+            logVerbose("\(#fileID) Sending event \(event)", environment: environment)
+            dispatcher.send(event)
+            return
+        }
+
+        guard let callbackURL = URL(string: hostedUIConfig.oauth.signOutRedirectURI),
+              let callbackURLScheme = callbackURL.scheme else {
+            let error = AuthenticationError.configuration(message: "Callback URL could not be retrieved")
+            let event = SignOutEvent(eventType: .signedOutFailure(error))
+            logVerbose("\(#fileID) Sending event \(event)", environment: environment)
+            dispatcher.send(event)
+            return
+        }
+
+        let signOutURI = hostedUIConfig.oauth
+            .signOutRedirectURI
+            .addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed)
+
+        let state = UUID().uuidString.lowercased()
+        var components = URLComponents()
+        components.scheme = "https"
+        components.path = "/logout"
+        components.host = hostedUIConfig.oauth.domain
+        components.queryItems = [
+            URLQueryItem(name: "client_id", value: hostedUIConfig.clientId),
+            URLQueryItem(name: "logout_uri", value: signOutURI),
+        ]
+
+        guard let logutURL = components.url else {
+            let error = AuthenticationError.configuration(message: "Could not create logout URL")
+            let event = SignOutEvent(eventType: .signedOutFailure(error))
+            logVerbose("\(#fileID) Sending event \(event)", environment: environment)
+            dispatcher.send(event)
+            return
+        }
+
+
+        let aswebAuthenticationSession = ASWebAuthenticationSession(
+            url: logutURL,
+            callbackURLScheme: callbackURLScheme,
+            completionHandler: { url, error in
+
+                if let url = url {
+                    let urlComponents = URLComponents(url: url, resolvingAgainstBaseURL: false)
+                    let queryItems = urlComponents?.queryItems
+
+                    if let error = queryItems?.first(where: { $0.name == "error" })?.value {
+                        let event = HostedUIEvent(eventType: .throwError(.hostedUI(.serviceMessage(error))))
+                        self.logVerbose("\(#fileID) Sending event \(event)", environment: environment)
+                        dispatcher.send(event)
+                        return
+                    }
+
+                    let event: SignOutEvent
+                    if self.signOutEvent.globalSignOut {
+                        event = SignOutEvent(eventType: .signOutGlobally(self.signInData))
+                    } else {
+                        event = SignOutEvent(eventType: .revokeToken(self.signInData))
+                    }
+                    self.logVerbose("\(#fileID) Sending event \(event.type)", environment: environment)
+                    dispatcher.send(event)
+                }
```

Comment:
can we do something as follows?
```
if let url = url { 
// Logic
} else if let error = error {
// Logic
} else {
// Throw an error
}
```
This would avoid the scenario where no event is sent from the Action.

Reply:
This is also handled in the next PR - https://github.com/aws-amplify/amplify-ios/pull/2030

Reply:
Logic is added inside - AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Support/HostedUI/HostedUIASWebAuthenticationSession.swift